### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by bobturneruk** [CITCOM-project/causcumber#14]
For some functions, `docstrings` have been used. This is great. `sphinx` and readthedocs should be configured to automatically build documentation based on these, using a `docs/` folder under the repository root. Docstrings should be added to all public functions, classes, etc. Example https://github.com/pykale/pykale

**Comment by bobturneruk** [comment]
A consistent docstring format needs to be decided on and stated in contributing guidelines.

**Comment by jmafoster1** [comment]
I agree about formatting. The differences probably come from different contributors using different editors which autogenerate their own skeleton formats. I assume at least some of the IDEs are customisable, so I suggest we use the format of the least customisable one. I use Spyder, which looks to support formats called "Numpy", "Google", and "Sphinx". I have no strong feelings either way.

**Comment by bobturneruk** [comment]
I'm most familiar with `sphinx`, so maybe go with that? Any interest in using typehints? This affects the decision a bit as sphinx will use these to document types.

**Comment by jmafoster1** [comment]
Cool, let's do that, then. I've switched Spyder to Sphinx mode, so hopefully that should be standard. To be honest, I sort of missed typehints becoming a thing. I'm in favour. I generally prefer strongly typed languages anyway.

**Comment by bobturneruk** [comment]
Say that quietly amongst the Python zealots.

We probably need to agree some contributing guidelines. I can draft (steal) some.
